interview-questions.md

```diff
@@ -4,64 +4,117 @@ Answer the following questions.
+                Otherwise, this refers to the global object. Using `this` in a class works similary to using it in a function. `This` acts as a
+                regular object.
 
+/**  Reference: **/ 
+                https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this
+                https://www.w3schools.com/js/js_this.asp
 
 2. What is React? Why would you use it?
 
   Your answer:
+                React is a library/collection of code snippets that allows developers to more easily create dynamic single page websites or applications.  
+                Developers are able to use JSX syntax which combines the structure of a normal html file with the code and behavior of a JS file all into one. 
 
-  Researched answer:
-
+  Researched answer: 
+                Using React allows for simpler oversight of the `View` aspect of an application. It creates a dynamic environment where data can be exchanged
+                and manipulated without having to reload the page. It is scalable , flexible and efficient when used to separate, update and render all 
+                of ones various components.
 
+/**  Reference: **/
+                https://reactjs.org/
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: 
+                The render method is required in a React class component
 
+  Researched answer: 
+                The lifecycle methods of a React component consist of , Mounting, Updating an Unmounting. In otherwords: 1. When an element is added to the DOM 
+                2. When the element is updated or changed and 3. When an element is removed. 
 
+/**  Reference: **/ 
+                https://www.w3schools.com/react/react_lifecycle.asp
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer:  As previously stated, JSX is a syntax extention the combines both the structure on a normal HTML file and the logic/behavior of a JS file all into one. Use of this syntactical 
+                sugar allows for both html and normal JS code to be written and displayed without having to use separate files for regular HTML and JS syntax. 
 
-  Researched answer:
+  Researched answer: 
+                The main difference would be that using JSX requires the return statement to return a single parent element. Failure to do so will cause a compling error. 
+                A fragment <> </> or the use of a single parent div circumvents this problem.
 
+/**  Reference: **/ 
+                https://www.geeksforgeeks.org/jsx-full-form/
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: 
+                Yarn is a software that manages dependency files.  When running Yarn , package.json , package-lock.json
+                and yarn.lock are modified using a React application  
 
-  Researched answer:
+  Researched answer: 
```

Comment:
It also adds several code snippets which are placed in node modules. Look in your react applications; you will see 100s of code added to support the build up of your project.